### PR TITLE
Parameterize MDSD account and namespace names

### PR DESCRIPTION
**Pull request by zacharyljones**
### Which issue this PR addresses:

[USER STORY 10098170 - Parameterize Geneva Namespace Targets in ARO RP](https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/10098170/)

### What this PR does / why we need it:

This PR allows the RP to read in additional mdsd logging parameters from RP Config files. These new parameters include mdsd log account name, namespace name, and metrics account name for both customer clusters and the rp itself. This was necessary because there are not separate mdsd endpoints for FF INT and PROD. These changes allow different accounts/namespaces to be targeted for different environments at the same mdsd endpoint.

RP Config has already been updated with the name parameters for each environment.

### Test plan for issue:

I tested this by deploying these RP changes to FF INT and verifying that the RP log and customer cluster data made it into the Geneva log accounts and namespaces specified in the FF INT config file. 

### Is there any documentation that needs to be updated for this PR?

No. RP Config has already been updated with an explanation of all new parameters.

### CODE REVIEW

File: pkg/env/prod.go [hunk 1]
```diff
@@ -91,8 +95,10 @@ func newProd(ctx context.Context, log *logrus.Entry) (*prod, error) {
 
 		fpClientID: os.Getenv("AZURE_FP_CLIENT_ID"),
 
-		clusterGenevaLoggingEnvironment:   os.Getenv("MDSD_ENVIRONMENT"),
+		clusterGenevaLoggingAccount:       os.Getenv("CLUSTER_MDSD_ACCOUNT"),
+		clusterGenevaLoggingNamespace:     os.Getenv("CLUSTER_MDSD_NAMESPACE"),
 		clusterGenevaLoggingConfigVersion: os.Getenv("CLUSTER_MDSD_CONFIG_VERSION"),
+		genevaLoggingEnvironment:          os.Getenv("MDSD_ENVIRONMENT"),
```
Comment (troy0820) [inline pkg/env/prod.go:101]: This name `genevaLoggingEnvironment` was `clusterGenevaLoggingEnvironment`.  Because it's the same env var, is there any reason why we are changing this name?

File: pkg/util/cluster/cluster.go [hunk 2]
```diff
@@ -113,6 +114,28 @@ func New(log *logrus.Entry, env env.Core, ci bool) (*Cluster, error) {
...
+		}
+
+		r, err := azure.ParseResourceID(c.ciParentVnet)
+		if err != nil {
+			return nil, err
+		}
+
+		c.ciParentVnetPeerings = network.NewVirtualNetworkPeeringsClient(env.Environment(), r.SubscriptionID, authorizer)
+	}
+
```
Comment (jewzaam) [inline pkg/util/cluster/cluster.go:138]: This whole block was removed in https://github.com/Azure/ARO-RP/pull/1573 along with other bits that were not pulled into this PR.  Please review the changes and make sure you've got a clean rebase with only intended changes.

File: test/e2e/setup.go [hunk 3]
```diff
@@ -201,7 +201,7 @@ func done(ctx context.Context) error {
 var _ = BeforeSuite(func() {
 	log.Info("BeforeSuite")
 
-	SetDefaultEventuallyTimeout(5 * time.Minute)
+	SetDefaultEventuallyTimeout(20 * time.Minute)
```
Comment (jewzaam) [inline test/e2e/setup.go:204]: If we want to bump this recommend we put in a separate PR.

**Review by jewzaam — CHANGES_REQUESTED** [review]
Review diff and prune those things you didn't intend to change (from https://github.com/Azure/ARO-RP/pull/1573).  Squash merge and re-push result.

File: pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go [hunk 4]
```diff
@@ -30,8 +30,12 @@ func ClusterChecksTypes() []status.ConditionType {
 type GenevaLoggingSpec struct {
 	// +kubebuilder:validation:Pattern:=`[0-9]+.[0-9]+`
 	ConfigVersion string `json:"configVersion,omitempty"`
-	// +kubebuilder:validation:Enum=DiagnosticsProd;Test
```
Comment (troy0820) [inline pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go:33]: I'm not sure why this is removed is the `kubebuilder:validation:Enum=DiagnosticsProd,Test;CaFairFax` is also on line 35

File: pkg/env/prod.go [hunk 5]
```diff
@@ -46,8 +46,10 @@ type prod struct {
 
 	clusterGenevaLoggingCertificate   *x509.Certificate
 	clusterGenevaLoggingPrivateKey    *rsa.PrivateKey
+	clusterGenevaLoggingAccount       string
 	clusterGenevaLoggingConfigVersion string
-	clusterGenevaLoggingEnvironment   string
+	clusterGenevaLoggingNamespace     string
+	genevaLoggingEnvironment          string
```
Comment (troy0820) [inline pkg/env/prod.go:52]: Same comment as below.  Naming should be consistent _(it references the same env variable)_

File: pkg/util/version/const.go [hunk 6]
```diff
@@ -12,8 +12,15 @@ import (
 const InstallArchitectureVersion = api.ArchitectureVersionV2
 
 const (
-	DevClusterGenevaLoggingEnvironment   = "Test"
-	DevClusterGenevaLoggingConfigVersion = "2.3"
+	DevClusterGenevaLoggingAccount       = "AROClusterLogs"
+	DevClusterGenevaLoggingConfigVersion = "2.4"
+	DevClusterGenevaLoggingNamespace     = "AROClusterLogs"
+	DevClusterGenevaMetricsAccount       = "AzureRedHatOpenShiftCluster"
+	DevGenevaLoggingEnvironment          = "Test"
```
Comment (troy0820) [inline pkg/util/version/const.go:19]: Should `DevGenevaLoggingEnvironment` be consistent with the naming from comments on the `pkg/env/env.go` prod struct? `DevClusterGenevaLoggingEnvironment`?

File: test/e2e/setup.go (hunk 3 above)
Comment (troy0820) [inline test/e2e/setup.go:204]: Increasing timeout? Why is this necessary?

**Review by troy0820 — CHANGES_REQUESTED** [review]
- [ ] Please squash commits as this unit of work doesn't need 35 commits
    - [ ] Vendor commit should be separate 
- [ ] Update unit test to include new _(generation of new mocks have been made and existing unit tests do not cover newly included config params, etc)_
- [ ] Address naming to be consistent with what is already in the prod struct of `pkg/env/`
- [ ] Address increase timeout with setup.go for testing

**Comment by zacharyljones** [comment]
Abandoning
